### PR TITLE
PR: Fix compatibility with pytest-qt >= 4

### PR DESCRIPTION
**Pull request by bnavigator**
This reverts #15782 and fixes the reason why pytest-qt was pinned: qtbot.addWidget() expects a widget as argument, but a few plugins were not widgets.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @bnavigator

### CODE REVIEW

**Review by ccordoba12 — COMMENTED** [review]
Thanks for your help @bnavigator! The only problem I see with this is that pytest-qt 4 requires PyQt 5.11, according to its Changelog:

https://github.com/pytest-dev/pytest-qt/blob/master/CHANGELOG.rst#400-2021-06-03

but we're still testing with 5.9.

**Comment by bnavigator** [comment]
But you can still use pytest-qt < 4 with this PR. It's just that the upper restriction is lifted.

**Comment by ccordoba12** [comment]
You're totally right. However, It seems a slow test is failing. I restarted the build a couple of times and it's still failing, so please take a look at that.

**Comment by bnavigator** [comment]
> so please take a look at that.

Done.

File: spyder/plugins/completion/tests/conftest.py [hunk 1]
```diff
@@ -13,6 +13,10 @@
 # Third party imports
 from qtpy.QtWidgets import QMainWindow
 
+# PyTest imports
+import pytest
+from pytestqt.qtbot import QtBot
+
```
Comment (ccordoba12) [inline spyder/plugins/completion/tests/conftest.py:19]: Please revert this change. Without it we can't run our tests individually due to the issue mentioned in the comment below with QtAwesome.

**Review by ccordoba12 — COMMENTED** [review]
Thanks for your help @bnavigator! One last comment, then this should be ready.

File: spyder/plugins/completion/tests/conftest.py (hunk 1 above)
Comment (bnavigator) [reply spyder/plugins/completion/tests/conftest.py]: Are you sure about this? Neither pytest nor pytestqt.qtbot are initializing anything which you haven't imported above the comment already.

I removed the last commit from HEAD.

**Review by ccordoba12 — APPROVED** [review]
@bnavigator, thanks for your help with this!

*Note*: The failures in our tests are unrelated to this and are fixed in PR #16026.
